src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -473,7 +474,183 @@ pass::EliminateSqueeze::EliminateSqueeze() {
         return false;
     };
 
-    auto m = std::make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    auto m = make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+namespace {
+bool check_squeeze(const shared_ptr<Node>& node) {
+    auto squeeze = dynamic_pointer_cast<ov::opset9::Squeeze>(node);
+    if (squeeze) {
+        auto axis = dynamic_pointer_cast<ov::opset9::Constant>(squeeze->input_value(1).get_node_shared_ptr());
+        if (axis) {
+            auto axis_val = axis->cast_vector<int64_t>();
+            if (axis_val.size() == 1 && axis_val[0] == 1) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+// Checks that Reshape actually equals to Squeeze op
+// 0, -1 values in the shape pattern are not allowed.
+bool check_reshape(const shared_ptr<Node>& node) {
+    auto reshape = dynamic_pointer_cast<ov::opset9::Reshape>(node);
+    if (reshape) {
+        auto shape_pattern = dynamic_pointer_cast<ov::opset9::Constant>(reshape->input_value(1).get_node_shared_ptr());
+        if (shape_pattern) {
+            auto pattern_val = shape_pattern->cast_vector<int64_t>();
+            bool is_valid_pattern = find(pattern_val.begin(), pattern_val.end(), 0) == pattern_val.end();
+            is_valid_pattern =
+                is_valid_pattern || find(pattern_val.begin(), pattern_val.end(), -1) == pattern_val.end();
+            if (!is_valid_pattern) {
+                return false;
+            }
+            pattern_val.insert(pattern_val.begin() + 1, 1);
+            auto in_shape = reshape->input_value(0).get_partial_shape();
+            if (in_shape.compatible(pattern_val)) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool check_axis(const shared_ptr<ov::opset9::Concat>& concat,
+                const shared_ptr<Node>& split,
+                bool is_special_case = false) {
+    auto axis = dynamic_pointer_cast<ov::opset9::Constant>(split->input_value(1).get_node_shared_ptr());
+    if (!axis) {
+        return false;
+    }
+    const auto& axis_val = axis->cast_vector<int64_t>();
+    if (axis_val.size() != 1 || axis_val[0] != concat->get_axis()) {
+        return false;
+    }
```

Comment:
There is an option that these axes are in different forms. One is positive and one is negative.

Reply:
done